linode_api4/objects/nodebalancer.py

```diff
@@ -303,3 +303,25 @@ def statistics(self):
                 "Unexpected response generating stats!", json=result
             )
         return MappedObject(**result)
+
+    def firewalls(self):
+        """
+        View Firewall information for Firewalls associated with this NodeBalancer.
+
+        API Documentation: https://www.linode.com/docs/api/nodebalancers/#nodebalancer-firewalls-list
+
+        :returns: A List of Firewalls of the Linode NodeBalancer.
+        :rtype: List[Firewall]
+        """
+        from linode_api4.objects import (  # pylint: disable=import-outside-toplevel
+            Firewall,
+        )
+
+        result = self._client.get(
+            "{}/firewalls".format(NodeBalancer.api_endpoint), model=self
+        )
+
+        return [
+            Firewall(self._client, firewall["id"])
+            for firewall in result["data"]
+        ]
```

Comment:
Is the list of firewall filterable or can be paginated? If so we can take in `*filters` param and probably can use `_get_and_filter()` to call it. It can return a paginated list of Firewall.

We can pass the `endpoint` param specifically: https://github.com/linode/linode_api4-python/blob/ddb0b9800b5a3b71cb0e470dd20d0597c120a3cd/linode_api4/linode_client.py#L456

```suggestion
     return self.client._get_and_filter(Firewall, *filters, "{}/firewalls".format(NodeBalancer.api_endpoint))
```

Reply:
While the endpoint is a listing endpoint with pagination, nodebalancers can only have one Firewall. I'm not opposed to this design and maybe that doesn't change anything. But the list will only ever be a list of 1.